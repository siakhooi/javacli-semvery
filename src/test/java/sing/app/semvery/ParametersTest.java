package sing.app.semvery;

import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;

@ExtendWith({SnapshotExtension.class})
class ParametersTest {

  private Expect expect;

  @Test
  void testPrintUsage() throws Exception {
    Parameters p = new Parameters();
    p.process(new String[] {"-h"});
    String text = tapSystemOut(() -> p.printUsage());
    assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
  }

  @Test
  void testProcess1() {
    Parameters p = new Parameters();
    p.process(new String[] {"-h"});
    assertEquals(true, p.help);
    assertEquals(false, p.version);
    assertEquals(0, p.mainParameters.size());
  }

  @Test
  void testProcess2() {
    Parameters p = new Parameters();
    p.process(new String[] {"-v"});
    assertEquals(false, p.help);
    assertEquals(true, p.version);
    assertEquals(0, p.mainParameters.size());
  }

  @Test
  void testProcess3() {
    Parameters p = new Parameters();
    p.process(new String[] {"abc", "cde"});
    assertEquals(false, p.help);
    assertEquals(false, p.version);
    assertEquals(2, p.mainParameters.size());
  }
}
