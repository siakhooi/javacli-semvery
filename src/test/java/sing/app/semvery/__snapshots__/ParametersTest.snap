sing.app.semvery.ParametersTest.testPrintUsage=[
Usage: semvery [options]
  Options:
    --operation, -o
      Operation, value = [isValid]
    --version, -v
      Display version
    --help, -h
      Display help
.
.
]